File: activities/creature-battle/csharp/creature-battle-solution/BattleArena.cs
namespace creature_battle_solution;

using System;
using System.Threading;

public class BattleArena
{
    /// <summary>
    /// Manages a fight between two creatures until one is defeated.
    /// This method is static because it doesn't need an instance of BattleArena to run.
    /// </summary>
    public static void Fight(Creature c1, Creature c2)
    {
        Console.WriteLine("--- BATTLE START! ---");
        c1.Display();
        c2.Display();
        Console.WriteLine("-----------------------");
        Thread.Sleep(1000); // Pause for drama

        int round = 1;

        // The main battle loop continues as long as both creatures are alive
        while (c1.IsAlive() && c2.IsAlive())
        {
            Console.WriteLine($"\n--- Round {round} ---");

            // Creature 1 attacks Creature 2
            c1.Attack(c2);
            if (!c2.IsAlive()) break; // Check if the defender was defeated

            Thread.Sleep(500);

            // Creature 2 attacks Creature 1
            c2.Attack(c1);
            if (!c1.IsAlive()) break; // Check if the defender was defeated

            Console.WriteLine("\nEnd of Round Status:");
            c1.Display();
            c2.Display();
            Thread.Sleep(1000);
            round++;
        }

        Console.WriteLine("\n--- BATTLE OVER! ---");

        // Determine and announce the winner
        if (c1.IsAlive())
        {
            Console.WriteLine($"{c1.Name} is the victor!");
        }
        else if (c2.IsAlive())
        {
            Console.WriteLine($"{c2.Name} is the victor!");
        }
        else
        {
            Console.WriteLine("Both creatures have been defeated! It's a draw.");
        }
    }
}

File: activities/creature-battle/java/creature-battle-solution/src/main/java/org/example/BattleArena.java
package org.example;

public class BattleArena {
    public static void fight(Creature creature1, Creature creature2) {
        System.out.println("--- BATTLE START ---");
        creature1.display();
        creature2.display();
        System.out.println("--------------------");

        int round = 1;
        while (creature1.isAlive() && creature2.isAlive()) {
            System.out.println("\n--- Round " + round + " ---");

            // Creature 1 attacks Creature 2
            creature1.attack(creature2);
            if (!creature2.isAlive()) break; // Check if the fight is over

            // Creature 2 attacks Creature 1
            creature2.attack(creature1);

            // Display status at the end of the round
            creature1.display();
            creature2.display();
            round++;
        }

        System.out.println("\n--- BATTLE OVER ---");
        if (creature1.isAlive()) {
            System.out.println(creature1.name + " is the winner!");
        } else if (creature2.isAlive()) {
            System.out.println(creature2.name + " is the winner!");
        } else {
            System.out.println("It's a draw!");
        }
    }
}